app/app-upgrade-client/src/components/nav/nav.tsx

```diff
@@ -0,0 +1,240 @@
+  menuLinks: {
+    [theme.breakpoints.down('sm')]: {
+      display: 'none'
+    }
+  },
+  menuButton: {
+    [theme.breakpoints.up('md')]: {
+      display: 'none'
+    }
+  },
+  menuButtonLink: {
+    marginRight: theme.spacing(2),
+    display: 'inline-block',
+    fontSize: '18px',
+    color: theme.palette.primary.contrastText,
+    '&:hover': {
+      cursor: 'pointer',
+      color: '#F5F5F5'
+    }
+  },
+  menuButtonUser: {
+    marginLeft: theme.spacing(4)
+  },
+  closeMenuButton: {
+    marginRight: 'auto',
+    marginLeft: 0
+  },
+  drawerPaper: {
+    width: drawerWidth,
+    [theme.breakpoints.up('md')]: {
+      display: 'none'
+    }
+  },
+  logoutLoginDrawerWrapper: {
+    padding: '8px 16px',
+
+    '& button': {
+      marginLeft: '0px !important'
+    }
+  }
+}));
+
+export const Nav: FC = () => {
+  const { pathname } = useLocation();
+  // const pathname = location.pathname;
+  const { user } = useSelector((state: RootState) => {
+    return {
+      user: state.auth.user
+    };
+  });
+  const [mobileMenuOpen, setMobileMenuOpen] = React.useState(false);
+  const classes = useStyles();
+  const dispatch = useDispatch();
+
+  function toggleMobileMenu() {
+    setMobileMenuOpen(!mobileMenuOpen);
+  }
+
+  const logoutUser = () => {
+    dispatch(signOutUser());
+  };
+
+  const logoutLogin = (
+    <>
+      {user ? (
+        <Link href="/" underline="none">
+          <Button
+            variant="contained"
+            color="secondary"
+            onClick={logoutUser}
+            className={classes.menuButtonUser}
+          >
+            Logout
+          </Button>
+        </Link>
+      ) : (
+        <Link href="/login" underline="none">
+          <Button
+            variant="contained"
+            color="secondary"
+            className={classes.menuButtonUser}
+          >
+            Login
+          </Button>
+        </Link>
+      )}
+    </>
+  );
+
+  const drawer = (
+    <div>
+      <List>
+        <ListItem button onClick={toggleMobileMenu}>
+          <Link href="/" underline="none">
+            <ListItemText primary="Home" />
+          </Link>
+        </ListItem>
+        {appLinks.map(({ label, url }) => (
+          <ListItem button key={label} onClick={toggleMobileMenu}>
+            <Link href={url} underline="none">
+              <ListItemText primary={label} />
+            </Link>
+          </ListItem>
+        ))}
+        {pathname !== '/login' && (
+          <>
+            <Divider />
+            <div className={classes.logoutLoginDrawerWrapper}>
+              {logoutLogin}
+            </div>
+          </>
+        )}
+      </List>
+    </div>
+  );
+
+  return (
+    <div>
+      <Drawer
+        variant="temporary"
+        anchor="right"
+        open={mobileMenuOpen}
+        onClose={toggleMobileMenu}
+        classes={{
+          paper: classes.drawerPaper
+        }}
+        ModalProps={{
+          keepMounted: true // Better open performance on mobile.
```

Comment:
We don't really care about mobile. This is really just a desktop SAAS app so I don't think we should make much (if any) effort to support it.

Reply:
done